app/code/Magento/Captcha/Observer/CheckContactUsFormObserver.php

```diff
@@ -45,60 +52,48 @@ class CheckContactUsFormObserver implements ObserverInterface
     private $dataPersistor;
 
     /**
-     * @param \Magento\Captcha\Helper\Data $helper
-     * @param \Magento\Framework\App\ActionFlag $actionFlag
-     * @param \Magento\Framework\Message\ManagerInterface $messageManager
-     * @param \Magento\Framework\App\Response\RedirectInterface $redirect
+     * @param Data $helper
+     * @param ActionFlag $actionFlag
+     * @param ManagerInterface $messageManager
+     * @param RedirectInterface $redirect
      * @param CaptchaStringResolver $captchaStringResolver
+     * @param DataPersistorInterface $dataPersistor
      */
     public function __construct(
-        \Magento\Captcha\Helper\Data $helper,
-        \Magento\Framework\App\ActionFlag $actionFlag,
-        \Magento\Framework\Message\ManagerInterface $messageManager,
-        \Magento\Framework\App\Response\RedirectInterface $redirect,
-        CaptchaStringResolver $captchaStringResolver
+        Data $helper,
+        ActionFlag $actionFlag,
+        ManagerInterface $messageManager,
+        RedirectInterface $redirect,
+        CaptchaStringResolver $captchaStringResolver,
+        DataPersistorInterface $dataPersistor
     ) {
         $this->_helper = $helper;
         $this->_actionFlag = $actionFlag;
         $this->messageManager = $messageManager;
         $this->redirect = $redirect;
         $this->captchaStringResolver = $captchaStringResolver;
+        $this->dataPersistor = $dataPersistor;
```

Comment:
Hello @Bartlomiejsz, thank you for your contribution!

Even if this change doesn't affect a public API, I would recommend to make it as backward compatible as possible, adding the new constructor parameter as last and optional (`DataPersistorInterface $dataPersisto = null`), with property initialized with object manager if the parameter is not passed.

e.g.

    $this->dataPersistor = $dataPersistor ?: ObjectManager::getInstance()->get(DataPeristorInterface::class);

Reply:
Hello @aleron75, I can change it if it is required, sure, but I believe this would be great to specify final requirements, because almost each PR is processed differently :)
I.e. https://github.com/magento/magento2/pull/26684 was accepted after modifying constructor to be not backward compatible, and in https://github.com/magento/magento2/pull/26269 I was even asked to modify it and remove object manager usage.
What are your thoughts?

Reply:
Let me check and get back to you soon, thank you for pointing it out

Reply:
Hello @Bartlomiejsz here I am; I verified and I apologize, you are right: we can accept backward-incompatible changes for non-api classes, so we could add new arguments without fallback to Object Manager.